rebar.config

```diff
@@ -35,9 +37,22 @@
 {erl_opts, [debug_info,
 	    report_warnings, {warn_format,1}, warn_export_vars,
 	    warn_obsolete_guard, warn_unused_import,
-	    warn_missing_spec, warn_untyped_record]}.
+	    warn_missing_spec, warn_untyped_record,
+	    warn_unused_vars,
+	    {platform_define, "^17", 'USE_ERL_SCAN_LINE'},
+	    {platform_define, "^19|^2", 'AT_LEAST_19'},
+	    {platform_define, "^2", 'AT_LEAST_20'},
+	    {platform_define, "^2[1-9]", 'AT_LEAST_21'}]}.
 
-{pre_hooks, [{"(linux|darwin|gnu)", compile, "make include/compile_flags.hrl"},
-             {"(freebsd|netbsd|openbsd|solaris|dragonfly)", compile, "gmake include/compile_flags.hrl"},
-	     {"win32", compile, "escript.exe write_compile_flags include/compile_flags.hrl"}]}.
 {post_hooks, [{clean, "./clean_doc.sh"}]}.
+
+{dialyzer, [{warnings, [unmatched_returns
+                       ]}
+           ,{plt_extra_apps, [erts, kernel, stdlib, compiler, crypto, syntax_tools]}
+           ]}.
+
+{profiles, [{test, [{erl_opts, [nowarn_missing_spec
```

Comment:
Why are these `nowarn_` options present here? They should not really be, since this PR is still using (our patched version of) `rebar`, isn't it?

Reply:
This is rebar3-specific and clears great many a warnings when running test.

Reply:
No, I do _not_ want to see that.  Instead, I want to see a pull request that does _one_ thing, and one thing only, and does it right.  This one is titled "Simplify build".  Let's have that.